ministryofjustice/cloud-platform-environments
# feat: 🎸 add github action to check terrform module versions

When PRs add new terraform modules, then it is important that the user is using the most up-to-date module version. The github action will fail if they aren't using the latest version.

To test, you can:

1) run `go test ./...` in `cmd/check-terraform-modules-are-latest/`

Or

2) build the image and run the action against an existing pr, here's an example pr which fails as it is using an old module version:

```bash
cd cmd/check-terraform-modules-are-latest
docker build -t gha-check-terraform-version . 
docker run --rm=true --env="REPO_NAME=cloud-platform-environments" --env="API_URL=https://modules.apps.live-2.cloud-platform.service.justice.gov.uk/" --env="PR_NUMBER=10294" gha-check-terraform-version
```

[This is the PR being referenced](https://github.com/ministryofjustice/cloud-platform-environments/pull/10294)
